src/models/ComponentStyle.js

```diff
@@ -60,11 +62,9 @@ export default (
       this.isStatic = !isHRMEnabled && isStaticRules(rules, attrs)
       this.componentId = componentId
 
-      if (!StyleSheet.master.hasInjectedComponent(componentId)) {
-        const placeholder =
-          process.env.NODE_ENV !== 'production' ? [`.${componentId} {}`] : []
-        StyleSheet.master.deferredInject(componentId, placeholder)
-      }
+      const placeholder =
+        process.env.NODE_ENV !== 'production' ? [`.${componentId} {}`] : []
+      StyleSheet.master.deferredInject(componentId, placeholder)
```

Comment:
Seems like we don’t need to call deferredInject here at all in production mode since there’s no rule to add?

Reply:
Seems like it should be fine to just wrap that whole `placeholder = bla; deferredInject()` block into a production check

Reply:
deferredInject is our secret sauce to keep the order of injection correct. It actually adds a marker and adds deferred rules to inject at a later point in time.

So it’s still super important